api/services/timelog.py

```diff
@@ -48,3 +51,44 @@ def get_user_tasks(self, user_id: int, offset: int, limit: int, start: date, end
             or []
         )
         return tasks
+
+    def create_task(self, task: TaskNew) -> Task:
+        new_task = Task(
+            date=task.date,
+            init=task.init,
+            end=task.end,
+            story=task.story,
+            description=task.description,
+            task_type=task.task_type,
+            updated_at=datetime.now(),
+            user_id=task.user_id,
+            project_id=task.project_id,
+        )
+        self.db.add(new_task)
+        self.db.commit()
+        self.db.refresh(new_task)
+        return new_task
+
+    def check_task_for_overlap(self, task: Task) -> TaskValidate:
+        validated_task = TaskValidate(is_task_valid=True, message="")
+        user_tasks_for_day = (
+            self.db.query(Task).where(Task.user_id == task.user_id, Task.date.between(task.date, task.date)).all() or []
```

Comment:
Why between and not `Task.date == task.date`?

Reply:
No good reason. I think I had probably been copying the query from the `get_user_tasks` and didn't adjust this part.